clippy_dev/src/main.rs

```diff
@@ -19,12 +19,17 @@ fn main() {
     let matches = App::new("Clippy developer tooling")
         .subcommand(
             SubCommand::with_name("update_lints")
-                .about("Update the lint list")
+                .about("Makes sure that:\n \
+                       * the lint count in README.md is correct\n \
+                       * the changelog contains markdown link references at the bottom\n \
+                       * all lints groups include the correct lints\n \
```

Comment:
NIT:
```suggestion
                       * all lint groups include the correct lints\n \
```

r=me with that change